mtp_noms_ops/apps/security/views/check.py

```diff
@@ -28,6 +30,44 @@ class CreditsHistoryListView(SecurityView):
     form_class = CreditsHistoryListForm
 
 
+class CheckAssignView(BaseFormView):
+    """
+    Modify assignment of check
+    """
+    form_class = AssignCheckToUserForm
+    id_kwarg_name = 'check_id'
+    object_context_key = 'check'
+    list_url = reverse_lazy('security:check_list')
+
+    def get_success_url(self):
+        return reverse('security:resolve_check', kwargs={'check_id': self.kwargs[self.id_kwarg_name]})
+
+    def get_form_kwargs(self):
+        form_kwargs = super().get_form_kwargs()
+        form_kwargs.update(
+            {
+                'request': self.request,
+                'object_id': self.kwargs[self.id_kwarg_name],
+            },
+        )
+        return form_kwargs
+
+    def get(self, request, *args, **kwargs):
+        return HttpResponseRedirect(self.get_success_url())
+
+    def form_valid(self, form):
+        if self.request.method == 'POST':
+            result = form.assign_or_unassign()
+            if not result:
+                messages.add_message(
```

Comment:
i don't think this will work. `CheckAssignView.form_valid` is only called _if_ the form was valid so adding an error message won't allow it show up anywhere. you can, however, use `messages.error(self.request, _('Credit could not be added to your list'))`